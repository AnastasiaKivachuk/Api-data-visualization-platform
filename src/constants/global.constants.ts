export const DATE_FORMAT = {
    DATE_TIME_NUMBER: 'DD.MM.YYYY, HH:mm A',
    DATE_WORD: 'MMM DD, YYYY',
};