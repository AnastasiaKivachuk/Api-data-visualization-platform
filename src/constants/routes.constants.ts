export const HOME_ROUTE = '/';

export const USERS_ROUTE = '/users';
export const USER_ROUTE = '/users/[id]';

export const LAUNCHES_ROUTE = '/launches';
export const LAUNCH_ROUTE = '/launches/[id]';