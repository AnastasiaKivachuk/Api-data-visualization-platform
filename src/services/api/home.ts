import { gql } from '@apollo/client';
import client from '../../../apollo-client';

export const getHome = () => client.query({
  query: gql`
        query HomeScheme {
            company {
                ceo
                employees
                founded
                founder
                links {
                    elon_twitter
                    website
                }
                name
                summary
            }
            ships {
                id
                image
                model
                name
                url
                year_built
            }
            rockets {
                wikipedia
                name
                id
                description
            }
        }
    `,
});
