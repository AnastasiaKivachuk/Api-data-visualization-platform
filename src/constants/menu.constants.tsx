import {HOME_ROUTE, LAUNCHES_ROUTE, USERS_ROUTE} from './routes.constants';
import GroupIcon from '@mui/icons-material/Group';
import HomeIcon from '@mui/icons-material/Home';
import RocketLaunchIcon from '@mui/icons-material/RocketLaunch';
import React from "react";

export const APP_BAR_ITEMS = [
    {
        link: HOME_ROUTE,
        text: '',
        icon: <HomeIcon/>
    },
    {
        link: USERS_ROUTE,
        text: 'Users',
        icon: <GroupIcon/>
    },
    {
        text: 'Launches',
        link: LAUNCHES_ROUTE,
        icon: <RocketLaunchIcon/>
    }
]