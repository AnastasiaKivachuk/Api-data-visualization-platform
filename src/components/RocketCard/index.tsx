import * as React from 'react';
import {useRouter} from "next/router";

import {Card as MuiCard} from '@mui/material';
import CardActions from '@mui/material/CardActions';
import CardContent from '@mui/material/CardContent';
import Button from '@mui/material/Button';
import Typography from '@mui/material/Typography';

import {RocketsInterface} from "../../interfaces/home.interface";

 const RocketCard = ({description, name, wikipedia}: RocketsInterface) => {
    const router = useRouter();
    const openUrl = () => router.push(wikipedia);
    return (
        <MuiCard sx={{maxWidth: 345}}>
            <CardContent>
                <Typography gutterBottom variant="h5" component="div">
                    {name}
                </Typography>
                <Typography variant="body2" color="text.secondary">
                    {description}
                </Typography>
            </CardContent>
            {wikipedia && <CardActions>
                <Button variant="outlined" size="small" onClick={openUrl}>Learn More</Button>
            </CardActions>
            }
        </MuiCard>
    );
}

export default RocketCard;