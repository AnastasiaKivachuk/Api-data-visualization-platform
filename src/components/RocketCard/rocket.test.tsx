import {shallow} from "enzyme";
import React from "react";

import {RocketsInterface} from "../../interfaces/home.interface";
import RocketCard from "./index";
import {HOME_ROUTE} from "../../constants/routes.constants";

const MockRocketData: RocketsInterface = {
    description: "The Falcon 1 was an expendable launch system privately developed and manufactured by SpaceX during 2006-2009. On 28 September 2008, Falcon 1 became the first privately-developed liquid-fuel launch vehicle to go into orbit around the Earth.",
    name: "Falcon 1",
    wikipedia: "https://en.wikipedia.org/wiki/Falcon_1"
};

const setUp = (props: RocketsInterface) => shallow(<RocketCard {...props} />);


jest.mock('next/router', () => ({
    useRouter() {
        return ({
            route: '/',
            pathname: '',
            query: '',
            asPath: '',
            push: jest.fn(),
            events: {
                on: jest.fn(),
                off: jest.fn()
            },
            beforePopState: jest.fn(() => null),
            prefetch: jest.fn(() => null)
        });
    },
}));

describe("RocketCard component", () => {

    beforeEach(() => {
        const useRouter = jest.spyOn(require("next/router"), "useRouter");
        useRouter.mockImplementation(() => ({
            route: HOME_ROUTE,
            pathname: '',
            query: '',
            asPath: '',
            push: jest.fn(),
            events: {
                on: jest.fn(),
                off: jest.fn()
            },
            beforePopState: jest.fn(() => null),
            prefetch: jest.fn(() => null)
        }));
    })

    it("should render RocketCard component", () => {
        const component = setUp(MockRocketData)
        expect(component).toMatchSnapshot();
    })

    it("check all elements with props", () => {
        const component = setUp(MockRocketData)
        const text = ['Falcon 1', "The Falcon 1 was an expendable launch system privately developed and manufactured by SpaceX during 2006-2009. On 28 September 2008, Falcon 1 became the first privately-developed liquid-fuel launch vehicle to go into orbit around the Earth."]
        expect(component.find('ForwardRef(Typography)').length).toBe(2);
        component.find('ForwardRef(Typography)').map((item, index) => {
            expect(item.text()).toBe(text[index]);
        })
    })

    it("check button link", () => {
        const component = setUp(MockRocketData)
        expect(component.find('ForwardRef(Button)').length).toBe(1);
        component.find('ForwardRef(Button)').simulate('click')
    })
})
