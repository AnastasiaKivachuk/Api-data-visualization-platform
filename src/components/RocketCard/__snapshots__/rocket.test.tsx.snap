// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RocketCard component should render RocketCard component 1`] = `
<ForwardRef(Card)
  sx={
    Object {
      "maxWidth": 345,
    }
  }
>
  <ForwardRef(CardContent)>
    <ForwardRef(Typography)
      component="div"
      gutterBottom={true}
      variant="h5"
    >
      Falcon 1
    </ForwardRef(Typography)>
    <ForwardRef(Typography)
      color="text.secondary"
      variant="body2"
    >
      The Falcon 1 was an expendable launch system privately developed and manufactured by SpaceX during 2006-2009. On 28 September 2008, Falcon 1 became the first privately-developed liquid-fuel launch vehicle to go into orbit around the Earth.
    </ForwardRef(Typography)>
  </ForwardRef(CardContent)>
  <ForwardRef(CardActions)>
    <ForwardRef(Button)
      onClick={[Function]}
      size="small"
      variant="outlined"
    >
      Learn More
    </ForwardRef(Button)>
  </ForwardRef(CardActions)>
</ForwardRef(Card)>
`;
